python_modules/dagster/dagster_tests/core_tests/resource_tests/pythonic_resources/test_binding.py

```diff
@@ -585,3 +587,36 @@ def my_asset(context, my_resource: MyResourceWithDefault):
     )
 
     assert executed["yes"]
+
+
+def test_late_binding_with_resource_defs() -> None:
+    class DummyDB:
+        def execute_query(self, query):
+            pass
+
+    @resource
+    def dummy_database_resource(init_context):
+        return DummyDB()
+
+    @op(required_resource_keys={"database"})
+    def op_requires_resources(context):
+        context.resources.database.execute_query("foo")
+
+    @job(resource_defs={"database": dummy_database_resource})
+    def do_database_stuff():
+        op_requires_resources()
+
+    @op
+    def simple_op():
+        pass
+
+    @job()
+    def simple_job():
+        simple_op()
+
+    # io_manager here will be bound to both jobs
+    # we need to make sure this doesn't invalidate the database resource
+    Definitions(
+        jobs=[do_database_stuff, simple_job],
+        resources={"io_manager": FilesystemIOManager()},
+    )
```

Comment:
could we also test the behavior when you provide the key at both the job level and the definitions level?